aiokafka/consumer.py

```diff
@@ -1,375 +1,623 @@
+
+        self._coordinator = GroupCoordinator(
+            self._client, self._subscription, loop=self._loop,
+            group_id=self._group_id,
+            heartbeat_interval_ms=self._heartbeat_interval_ms,
+            retry_backoff_ms=self._retry_backoff_ms,
+            enable_auto_commit=self._enable_auto_commit,
+            auto_commit_interval_ms=self._auto_commit_interval_ms,
+            assignors=self._partition_assignment_strategy)
+
+        yield from self._coordinator.ensure_active_group()
+
+    def assign(self, partitions):
+        """Manually assign a list of TopicPartitions to this consumer.
+
+        Arguments:
+            partitions (list of TopicPartition): assignment for this instance.
+
+        Raises:
+            IllegalStateError: if consumer has already called subscribe()
+
+        Warning:
+            It is not possible to use both manual partition assignment with
+            assign() and group assignment with subscribe().
+
+        Note:
+            This interface does not support incremental assignment and will
+            replace the previous assignment (if there was one).
+
+        Note:
+            Manual topic assignment through this method does not use the
+            consumer's group management functionality. As such, there will be
+            no rebalance operation triggered when group membership or cluster
+            and topic metadata change.
+        """
+        self._subscription.assign_from_user(partitions)
+        self._client.set_topics([tp.topic for tp in partitions])
 
-log = logging.getLogger("aiokafka")
+    def assignment(self):
+        """Get the TopicPartitions currently assigned to this consumer.
 
+        If partitions were directly assigned using assign(), then this will
+        simply return the same partitions that were previously assigned.
+        If topics were subscribed using subscribe(), then this will give the
+        set of topic partitions currently assigned to the consumer (which may
+        be none if the assignment hasn't happened yet or if the partitions are
+        in the process of being reassigned).
 
-class AIOConsumer(object):
+        Returns:
+            set: {TopicPartition, ...}
+        """
+        return self._subscription.assigned_partitions()
 
-    AUTO_COMMIT_MSG_COUNT = 100
-    AUTO_COMMIT_INTERVAL = 5000
-    FETCH_DEFAULT_BLOCK_TIMEOUT = 1
-    FETCH_BUFFER_SIZE_BYTES = 4096
-    MAX_FETCH_BUFFER_SIZE_BYTES = FETCH_BUFFER_SIZE_BYTES * 8
+    @asyncio.coroutine
+    def stop(self):
+        """Close the consumer, waiting indefinitely for any needed cleanup."""
+        if self._closed:
+            return
+        log.debug("Closing the KafkaConsumer.")
+        self._closed = True
+        if self._coordinator:
+            yield from self._coordinator.close()
+        if self._fetcher:
+            yield from self._fetcher.close()
+        self._client.close()
+        try:
+            self._key_deserializer.close()
```

Comment:
We documented, that deserializer is a callable. Nothing about close method there. Maybe it's some backward-compatibility for python-kafka, and we don't need it?